buildbot/start_script.sh

```diff
@@ -104,73 +107,106 @@ sudo service stackdriver-agent start
 update-alternatives --install "/usr/bin/ld" "ld" "/usr/bin/ld.gold" 20
 update-alternatives --install "/usr/bin/ld" "ld" "/usr/bin/ld.bfd" 10
 
+#mkdir -p $BOT_DIR/.ccache
+#cache_dir = $BOT_DIR/.ccache
+mkdir -p /var/lib/buildbot/.ccache
+chown -R buildbot:buildbot /var/lib/buildbot/.ccache
+cat <<EOF >/var/lib/buildbot/.ccache/ccache.conf
+max_size = 50.0G
+compression = true
+EOF
+
 # Generate Debian image for QEMU bot.
-mkdir -p $BOT_DIR/qemu_image
 (
   set -ux
+  rm -rf $BOT_DIR/qemu_image
+  mkdir -p $BOT_DIR/qemu_image
   cd $BOT_DIR/qemu_image
 
   SLEEP=0
   for i in `seq 1 5`; do
     sleep $SLEEP
     SLEEP=$(( SLEEP + 10))
-
-    rm -rf *
-    (
-      curl "https://raw.githubusercontent.com/google/sanitizers/master/hwaddress-sanitizer/create_qemu_image.sh" \
-        | bash
-    ) && exit 0
+    ${SCRIPT_DIR}/../hwaddress-sanitizer/create_qemu_image.sh && exit 0
   done
   exit 1
 ) || $ON_ERROR
 
-SERVICE_NAME=buildbot-worker@b.service
-[[ -d /var/lib/buildbot/workers/b ]] || ln -s $BOT_DIR /var/lib/buildbot/workers/b
-
-systemctl enable $SERVICE_NAME
-systemctl set-property $SERVICE_NAME TasksMax=100000
-
-systemctl stop $SERVICE_NAME || true
-while pkill buildbot-worker; do sleep 5; done;
-
-rm -f /b/buildbot.tac
-buildbot-worker create-worker -f --allow-shutdown=signal $BOT_DIR lab.llvm.org:$MASTER_PORT \
-  "sanitizer-$(hostname | cut -d '-' -f2)" \
-  "$(gsutil cat gs://sanitizer-buildbot/buildbot_password)"
-
-
-echo "Vitaly Buka <vitalybuka@google.com>" > $BOT_DIR/info/admin
-
-{
-  echo "How to reproduce locally: https://github.com/google/sanitizers/wiki/SanitizerBotReproduceBuild"
-  echo
-  uname -a | head -n1
-  date
-  cmake --version | head -n1
-  g++ --version | head -n1
-  ld --version | head -n1
-  lscpu
-} > $BOT_DIR/info/host
-
-mkdir -p $BOT_DIR/.ccache
-mkdir -p /var/lib/buildbot/.ccache
-cat <<EOF >/var/lib/buildbot/.ccache/ccache.conf
-max_size = 20.0G
-compression = true
-cache_dir = $BOT_DIR/.ccache
-EOF
-
-chown -R buildbot:buildbot $BOT_DIR
-systemctl daemon-reload
-systemctl start $SERVICE_NAME
-systemctl status $SERVICE_NAME
-
-sleep 30
-cat $BOT_DIR/twistd.log
-grep "worker is ready" $BOT_DIR/twistd.log || $ON_ERROR
-
-# GCE can restart instance after 24h in the middle of the build.
-# Gracefully restart before that happen.
-sleep 72000
-while pkill -SIGHUP buildbot-worker; do sleep 5; done;
-$ON_ERROR
```

Comment:
Yes, I am not sure if it's was useful from the beginning.
There was assumption that VM will be alive at least this time, but there is no such guaranty from GCE.
It can be replicated with new logic, but I am not convinced we need this.